docs/vectors/c-source/sealbox_test_vectors.c

```diff
@@ -0,0 +1,150 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Test vector generator/checker for libsodium's box_seal APIs
+ * to build in a unix-like environment, use a command line like
+ * $ cc sealbox_test_vectors.c -I${IPATH} -L${LPATH} -lsodium -o sealbox_test_vectors
+ * with IPATH and LPATH defined to respectively point to libsodium's include path
+ * and to the directory containing the link library libsodium.a or libsodium.o
+ *
+ */
+#include <stdio.h>
+#include <string.h>
+#include <sodium.h>
+
+int checkone (char *hxsecret, char *hxpub, char *hxplaintext, char *hxencrypted) {
+
+    int pklen = crypto_box_PUBLICKEYBYTES;
+    int sklen = crypto_box_SECRETKEYBYTES;
+    int txtlen = strlen (hxplaintext) / 2;
+    int encrlen = strlen (hxencrypted) / 2;
+
+    char *skr = sodium_malloc (sklen);
+    char *pub = sodium_malloc (pklen);
+    char *txt = sodium_malloc (txtlen);
+    char *crpt = sodium_malloc (encrlen);
+    char *outp = sodium_malloc (txtlen);
+
+    int rs = sodium_hex2bin (skr, sklen, hxsecret, 2 * sklen,
+                             NULL, NULL, NULL);
+    rs |= sodium_hex2bin (pub, pklen, hxpub, 2 * pklen, NULL, NULL, NULL);
+    rs |= sodium_hex2bin (txt, txtlen, hxplaintext, strlen (hxplaintext),
+                          NULL, NULL, NULL);
+    rs |= sodium_hex2bin (crpt, encrlen, hxencrypted, strlen (hxencrypted),
+                          NULL, NULL, NULL);
+
+    if (rs == 0)
+        rs = crypto_box_seal_open (outp, crpt, encrlen, pub, skr);
+    if (rs == 0)
+        rs = sodium_memcmp (outp, txt, txtlen);
+
+    sodium_free (crpt);
+    sodium_free (txt);
+    sodium_free (skr);
+    sodium_free (pub);
+
+    return rs;
+}
+
+void gentestline (int minmsglen, int maxmsglen) {
+
+    int pklen = crypto_box_PUBLICKEYBYTES;
+    int sklen = crypto_box_SECRETKEYBYTES;
+    int txtlen = minmsglen + randombytes_uniform (maxmsglen - minmsglen + 1);
+    int encrlen = txtlen + crypto_box_SEALBYTES;
+
+    char *skr = sodium_malloc (sklen);
+    char *pub = sodium_malloc (pklen);
+    char *txt = sodium_malloc (txtlen);
+    char *crpt = sodium_malloc (encrlen);
+
+    crypto_box_keypair (pub, skr);
+    randombytes_buf (txt, txtlen);
+
+    crypto_box_seal (crpt, txt, txtlen, pub);
+
+    char *hskr = sodium_malloc (sklen * 2 + 1);
+    char *hpub = sodium_malloc (pklen * 2 + 1);
+    char *htxt = sodium_malloc (txtlen * 2 + 1);
+    char *hkrp = sodium_malloc (encrlen * 2 + 1);
+
+    sodium_bin2hex (hskr, sklen * 2 + 1, skr, sklen);
+    sodium_bin2hex (hpub, pklen * 2 + 1, pub, pklen);
+    sodium_bin2hex (htxt, txtlen * 2 + 1, txt, txtlen);
+    sodium_bin2hex (hkrp, encrlen * 2 + 1, crpt, encrlen);
+
+    printf ("%s\t%s\t%s\t%s\n", hskr, hpub, htxt, hkrp);
+}
+
+int main (int argc, char **argv) {
+/*
+ * If called without any argument, the resulting executable will
```

Comment:
Since this is meant to be run with specific arguments, maybe add "example" commands that have correct test vectors as arguments, and include links to source of the test vectors. If the driver has no other uses, maybe explain how to verify SealedBox custom test vectors with this application: what inputs and outputs to expect.